### ISSUE TITLE
Use functional options

### ISSUE THREAD

**Issue by adammck** [NYTimes/gziphandler#43]
Dear Lazyweb: It would be nice if our API looked less like

    gz.NewGzipLevelAndMinSize(1, 1024)(myHandler)

and more like

    gz.New(myHandler, gz.CompressionLevel(1), gz.MinSize(1024))

See [Functional options for friendly APIs](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis). I'm open to other solutions, too.

**Comment by jprobinson** [comment]
After playing around with more and more go-kit, I've been leaning towards the latter lately.

**Comment by tmthrgd** [comment]
This was fixed with the merging of #51.

---

The main difference was that the options take the form:
```
wrap, _ := gz.GzipHandlerWithOpts(gz.CompressionLevel(1), gz.MinSize(1024))
wrap(myHandler)
```
instead of:
```
gz.New(myHandler, gz.CompressionLevel(1), gz.MinSize(1024))
```

Either this should be closed or considered a request to add the `New` function described above.
